contracts/voting/src/lib.rs

```diff
@@ -74,13 +56,18 @@ impl ProposalVotingContract {
     /// - `admin` - The address that can create proposals.
     /// - `voting_period_secs` - The default number of seconds of proposals lifetime for new proposals.
     /// - `target_approval_rate_bps` - The default required approval rate in basic points for new proposals.
-    /// - `total_voters` - The default max number of voters for new proposals.
+    /// - `participation` - The default max number of participation for new proposals.
+    /// - `admin_mode` - Certain functions like `voting` are open for anyone who wants to invoke them. This
+    /// doable, but if we are using this contract as dependency of another contract and we are interested in
+    /// restricting all operations to be only performed by the Admin address (see admin params), this should
+    /// be set to true.
     pub fn init(
         env: Env,
         admin: Address,
         voting_period_secs: u64,
         target_approval_rate_bps: u32,
-        total_voters: u32,
+        participation: u128,
+        admin_mode: bool,
     ) {
         let storage = env.storage().persistent();
 
```

Comment:
I think you suggested we use:
```
if storage
            .get::<_, ()>(&DataKey::AlreadyInitialized)
            .is_some()
```
instead of
```
if storage
            .get::<_, bool>(&DataKey::AlreadyInitialized)
            .is_some()
```

Reply:
Fixed at f191d321756d89adc47dca7973520dd7dc293c5f